core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -58,11 +63,148 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
 
   @Before
   override def setUp() {
-    super.setUp
+    super.setUp()
     client = AdminClient.createSimplePlaintext(this.brokerList)
     TestUtils.createTopic(this.zkUtils, topic, 2, serverCount, this.servers)
   }
 
+  @After
+  override def tearDown() {
+    client.close()
+    super.tearDown()
+  }
+
+  @Test
+  def testPurgeDataBeforeWithSeekToBeginning() {
+    val consumer = consumers.head
+
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    consumer.seekToBeginning(Collections.singletonList(tp))
+    assertEquals(0L, consumer.position(tp))
+
+    client.purgeDataBefore(Map((tp, 5L))).get()
+    consumer.seekToBeginning(Collections.singletonList(tp))
+    assertEquals(5L, consumer.position(tp))
+
+    client.purgeDataBefore(Map((tp, PurgeRequest.HIGH_WATERMARK))).get()
+    consumer.seekToBeginning(Collections.singletonList(tp))
+    assertEquals(10L, consumer.position(tp))
+  }
+
+  @Test
+  def testConsumeAfterPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    assertEquals(10, consumer.poll(1000).count())
+
+    client.purgeDataBefore(Map((tp, 3L))).get()
+    consumer.seek(tp, 1)
+    assertEquals(7, consumer.poll(1000).count())
+
+    client.purgeDataBefore(Map((tp, 8L))).get()
+    consumer.seek(tp, 1)
+    assertEquals(2, consumer.poll(1000).count())
+  }
+
+  @Test
+  def testLogStartOffsetAfterPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    client.purgeDataBefore(Map((tp, 3L))).get()
+
+    for (i <- 0 until serverCount)
+      assertEquals(3, servers(i).replicaManager.getReplica(tp).get.logStartOffset)
+  }
+
+  @Test
+  def testLogStartOffsetAfterAsyncPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    client.purgeDataBefore(Map((tp, 3L)))
+    Thread.sleep(1000)
```

Comment:
It is no longer needed after we add a dedicated thread in AdminClient to do `poll()`. Removed now.